e2e/cypress/integration/messaging/message_reply_too_long_spec.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Message Reply too long', () => {
+    before(() => {
+        // # Login and navigate to town-square
+        cy.toMainChannelView('user-1');
+
+        // # Post a new message to ensure there will be a post to click on
+        cy.postMessage('Hello ' + Date.now());
+    });
+
+    it('M18689 - ""Message too long" warning text"', () => {
+        // # Click "Reply"
+        cy.getLastPostId().then((postId) => {
+            cy.clickPostCommentIcon(postId);
+        });
+
+        // # Enter valid text into RHS
+        const replyValid = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ';
+        cy.get('#reply_textbox').clear().should('be.visible').type(replyValid).type('{enter}');
+
+        // * Check no warning
+        cy.get('.post-error').should('not.exist');
+
+        // # Enter too long text into RHS
+        const replyTooLong = replyValid.repeat((16383 / replyValid.length) + 1);
+        cy.get('#reply_textbox').invoke('val', replyTooLong).trigger('input');
+
+        // * Check warning doesn't overlap textbox
+        cy.get('.post-error').should('be.visible');
+        cy.get('#reply_textbox').should('be.visible');
+
+        // # Type "enter" into RHS
+        cy.get('#reply_textbox').type('{enter}');
+
+        // * Check warning highlight and doesn't overlap textbox
+        cy.get('.post-error').should('be.visible').should('have.class', 'animation--highlight');
```

Comment:
Please add assertion on error message 
```
and('have.text', 'Your message is too long. Character count: 16416/16383')
```

nit: For readability, you may use/chain `.add` after the first `.should`.

Reply:
`.and` should be the correct chainer, btw.